File: architecture/clean-architecture/javascript/interface-adapters/controllers/Controller.js
import Repository from '../repositories/Repository.js';
import Presenter from '../presenters/Presenter.js';
import Create from '../../core/use-cases/Create.js';
import Read from '../../core/use-cases/Read.js';
import ReadOne from '../../core/use-cases/ReadOne.js';
import Update from '../../core/use-cases/Update.js';
import Delete from '../../core/use-cases/Delete.js';
import DB from '../../infrastructure/db.js';

const db = new DB();
const repo = new Repository(db);
const presenter = new Presenter();

function create(paramObj) {
    const useCase = new Create(repo, presenter);
    return useCase.execute(paramObj);
}

function read(paramObj) {
    const useCase = new Read(repo, presenter);
    return useCase.execute(paramObj);
}

function readOne(paramObj) {
    const useCase = new ReadOne(repo, presenter);
    return useCase.execute(paramObj);
}

function update(paramObj) {
    const useCase = new Update(repo, presenter);
    return useCase.execute(paramObj);
}

function remove(paramObj) {
    const useCase = new Delete(repo, presenter);
    return useCase.execute(paramObj);
}

export default {
    create,
    read,
    readOne,
    update,
    remove
}

File: architecture/clean-architecture/java/interfaceadapters/controllers/Controller.java
package interfaceadapters.controllers;

import java.util.List;
import core.entities.Document;
import core.entities.Response;
import core.entities.ParamObj;
import core.usecases.Create;
import core.usecases.Read;
import core.usecases.ReadOne;
import core.usecases.Update;
import infrastructure.DB;
import core.usecases.Delete;
import interfaceadapters.presenters.Presenter;
import interfaceadapters.repositories.Repository;

public class Controller {
    private final Repository repo;

    public Controller(DB db) {
        repo = new Repository(db);
    }
    public Response<Document> create(ParamObj params) {
        Presenter<Document> presenter = new Presenter<Document>();
        Create useCase = new Create(repo, presenter);
        return useCase.execute(params);
    }
    public Response<List<Document>> read(ParamObj params) {
        Presenter<List<Document>> presenter = new Presenter<List<Document>>();
        Read useCase = new Read(repo, presenter);
        return useCase.execute(params);
    }
    public Response<Document> readOne(ParamObj params) {
        Presenter<Document> presenter = new Presenter<Document>();
        ReadOne useCase = new ReadOne(repo, presenter);
        return useCase.execute(params);
    }
    public Response<Document> update(ParamObj params) {
        Presenter<Document> presenter = new Presenter<Document>();
        Update useCase = new Update(repo, presenter);
        return useCase.execute(params);
    }
    public Response<Document> delete(ParamObj params) {
        Presenter<Document> presenter = new Presenter<Document>();
        Delete useCase = new Delete(repo, presenter);
        return useCase.execute(params);
    }
}
